### ISSUE TITLE
Second layer of filtering after returning from `FileSearchProviderNew`'s `provideFileSearchResults(...)`

### ISSUE THREAD

**Issue by joshspicer** [microsoft/vscode#226912]
Testing #226668

I suspect here that I am just not too well-versed in how these APIs work and this is as-expected, but thought i'd ask here anyway :) 

For this example I have hardcoded a single file to always return from `provideFileSearchResults(...)`

```ts
class mySearchProvider implements vscode.FileSearchProviderNew {
	provideFileSearchResults(pattern: string, options: vscode.FileSearchProviderOptions, token: vscode.CancellationToken): vscode.ProviderResult<vscode.Uri[]> {
		return [
			vscode.Uri.parse('memfs:/josh.txt'),
		]
	}
}
```

As expected, when I search for any substring of `josh` I see it pop up in the quick open search window

![Image](https://github.com/user-attachments/assets/c96026f4-785e-402c-a58e-3d82c0c2536c)

As expected, i've made this utterly useless since it won't actually show me the files in my project.  But in this case, I still expected to see `josh.txt` since it was hardcoded (as shown above).

![Image](https://github.com/user-attachments/assets/9bb7379c-4f1f-4c6f-90c9-fdfff5c85a1d)

Again, likely a misunderstanding on my part of how the API works. I thought this API was the "source of truth" for providing search results, yet my hardcoding of the value is doing some filtering outside of my (apparent) control.

**Comment by andreamah** [comment]
Yes, when you use the quick open, there is some post-processing that applies to the results. This happens at two points:
- after getting the result from the provider, we re-apply the glob patterns for include/exclude because there may be some special syntax of the glob patterns itself that require special interpretation (sibling clauses, ie: https://github.com/microsoft/vscode/blob/2e4a043ef387856c67ac934c36ca67073f56624d/.vscode/settings.json#L11-L13). This will be present regardless of whether you use the picker or an API to find the files.
- When using the picker, you get extra fuzzy filtering of the results. The file provider does not inherently provide any ordering or indication why it's a match, so it's sort of the picker's way of re-applying logic that the provider should have already done.

I can keep this issue open to track the documentation of this, though. It is confusing.
